core/src/main/scala-2.13+/scalaz/std/Map.scala

```diff
@@ -0,0 +1,190 @@
+
+  implicit def mapOrder[K: Order, V: Order]: Order[Map[K, V]] =
+    new Order[Map[K, V]] with MapEqual[K, V] {
+      def OK = Order[K]
+      def OV = Equal[V]
+      def order(x: Map[K, V], y: Map[K, V]): Ordering = {
+        import vector._
+        import anyVal._
+        import tuple._
+        implicit val ok = Order[K].toScalaOrdering
+        Semigroup[Ordering]
+         .append(Order[Int].order(x.size, y.size),
+                 Order[Vector[(K, V)]]
+                  .order(x.toVector.sortBy((_:(K,V))._1),
+                         y.toVector.sortBy((_:(K,V))._1)))
+      }
+    }
+}
+
+trait MapFunctions {
+  /** Vary the value of `m get k`. */
+  final def alter[K, A](m: Map[K, A], k: K)(f: (Option[A] => Option[A])): Map[K, A] =
+    f(m get k) map (x => m + ((k, x))) getOrElse (m - k)
+
+  /** Like `intersectWith`, but tell `f` about the key. */
+  final def intersectWithKey[K, A, B, C](m1: Map[K, A], m2: Map[K, B])(f: (K, A, B) => C): Map[K, C] = m1 collect {
+    case (k, v) if m2 contains k => k -> f(k, v, m2(k))
+  }
+
+  /** Collect only elements with matching keys, joining their
+    * associated values with `f`.
+    */
+  final def intersectWith[K, A, B, C](m1: Map[K, A], m2: Map[K, B])(f: (A, B) => C): Map[K, C] =
+    intersectWithKey(m1, m2)((_, x, y) => f(x, y))
+
+  /** Exchange keys of `m` according to `f`.  Result may be smaller if
+    * `f` maps two or more `K`s to the same `K2`, in which case the
+    * resulting associated value is an arbitrary choice.
+    */
+  final def mapKeys[K, K2, A](m: Map[K, A])(f: K => K2): Map[K2, A] =
+    m map {case (k, v) => f(k) -> v}
+
+  /** Like `unionWith`, but telling `f` about the key. */
+  final def unionWithKey[K, A](m1: Map[K, A], m2: Map[K, A])(f: (K, A, A) => A): Map[K, A] = {
+    val diff = m2 -- m1.keySet
+    val aug = m1 transform {
+      case (k, v) => if (m2 contains k) f(k, v, m2(k)) else v
+    }
+    aug ++ diff
+  }
+
+  /** Union, resolving collisions with `f`, where the first arg is
+    * guaranteed to be from `m1`, the second from `m2`.
+    *
+    * @note iff `f` gives rise to a [[scalaz.Semigroup]], so does
+    *       `unionWith(_, _)(f)`.*/
+  final def unionWith[K, A](m1: Map[K, A], m2: Map[K, A])(f: (A, A) => A): Map[K, A] =
+    unionWithKey(m1, m2)((_, x, y) => f(x, y))
+
+  /** As with `Map.updated`, but resolve a collision with `f`.  The
+    * first argument is guaranteed to be from `m1`.
+    */
+  final def insertWith[K, A](m1: Map[K, A], k: K, v: A)(f: (A, A) => A): Map[K, A] =
+    if(m1 contains k) m1 + ((k, f(m1(k), v))) else m1 + ((k, v))
+
+  /** Grab a value out of Map if it's present. Otherwise evaluate
+    * a value to be placed at that key in the Map.
+    */
+  final def getOrAdd[F[_], K, A](m: Map[K, A], k: K)(fa: => F[A])(implicit F: Applicative[F]): F[(Map[K, A], A)] =
+    (m get k).fold(F.map(fa)(a => (m + ((k, a)), a)))(a => F.point((m, a)))
+}
+
+object map extends MapInstances with MapFunctions
```

Comment:
Out of curiosity, what was incompatible with the new collections in the original `Map.scala` file?

Reply:
OK so now the second question is: it looks like the new version should work with both 2.12 and 2.13, no? Or maybe some features are missing from it?